samples/cellular/nrf_cloud_multi_service/src/cloud_connection.c

```diff
@@ -251,51 +251,16 @@ static bool connect_cloud(void)
  */
 static void update_shadow(void)
 {
-#if !defined(CONFIG_NRF_CLOUD_COAP) || defined(CONFIG_COAP_SHADOW)
-	static bool updated;
-	int err;
-	struct nrf_cloud_svc_info_fota fota_info = {
-		.application = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_APPLICATION),
-		.bootloader = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_BOOTLOADER),
-		.modem = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_MODEM_DELTA),
-		.modem_full = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_MODEM_FULL)
-	};
-	struct nrf_cloud_svc_info_ui ui_info = {
-		.gnss = location_tracking_enabled(),
-		.temperature = IS_ENABLED(CONFIG_TEMP_TRACKING),
-		.log = nrf_cloud_is_text_logging_enabled(),
-		.dictionary_log = nrf_cloud_is_dict_logging_enabled()
-	};
-	struct nrf_cloud_svc_info service_info = {
-		.fota = &fota_info,
-		.ui = &ui_info
-	};
-	struct nrf_cloud_device_status device_status = {
-		/* Modem info is sent automatically since CONFIG_NRF_CLOUD_SEND_DEVICE_STATUS
-		 * is enabled, so it can be skipped here.
-		 */
-		.modem = NULL,
-		.svc = &service_info,
-		.conn_inf = NRF_CLOUD_INFO_NO_CHANGE
-	};
-
-	if (updated) {
-		return; /* It is not necessary to do this more than once per boot. */
-	}
-	LOG_DBG("Updating shadow");
-#if defined(CONFIG_NRF_CLOUD_MQTT)
-	err = nrf_cloud_shadow_device_status_update(&device_status);
-#elif defined(CONFIG_NRF_CLOUD_COAP)
-	err = nrf_cloud_coap_shadow_device_status_update(&device_status);
-#endif
-
-	if (err) {
-		LOG_ERR("Failed to update device shadow, error: %d", err);
-	} else {
-		LOG_DBG("Updated shadow");
-		updated = true;
-	}
-#endif /* !defined(CONFIG_NRF_CLOUD_COAP) || defined(CONFIG_COAP_SHADOW) */
+	/* The nRF Cloud library will automatically set various info sections
+	 * in the device's shadow if enabled by the configuration;
+	 * see config NRF_CLOUD_SEND_SHADOW_INFO.
+	 *
+	 * If the configuration-based settings are not applicable to your application
+	 * the config option(s) should be adjusted and the info sections should be sent
+	 * here;
+	 * see function nrf_cloud_shadow_device_status_update() for MQTT and
+	 * nrf_cloud_coap_shadow_device_status_update() for CoAP.
```

Comment:
Nice! I think we can just remove this function completely, then, right?

Reply:
You could put a comment like this where the call to `update_shadow();` currently is

> The nRF Cloud library will automatically update the device's shadow based on the build configuration. See config NRF_CLOUD_SEND_SHADOW_INFO for details. 


Then I think the `NRF_CLOUD_SEND_SHADOW_INFO` docs can mention `nrf_cloud_shadow_device_status_update/nrf_cloud_coap_shadow_device_status_update`

Reply:
i left the `update_shadow()` function just for instructional purposes, along with adding the long comment.
if you prefer your suggestion, i will remove it and use the short comment.

Reply:
Personally I think it would be easier for a total beginner to understand the in-line comment as opposed to the instructional function, so that would be my preference